fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/SimplifyRange.java

```diff
@@ -498,41 +494,53 @@ private UnknownValue(ExpressionRewriteContext context, Expression expr) {
             mergeExprOp = null;
         }
 
-        public UnknownValue(ExpressionRewriteContext context,
-                List<ValueDesc> sourceValues, Expression exprForNonNull, BinaryOperator<Expression> mergeExprOp) {
-            super(context, sourceValues.get(0).reference, exprForNonNull);
+        public UnknownValue(ExpressionRewriteContext context, Expression toExpr,
+                List<ValueDesc> sourceValues, BinaryOperator<Expression> mergeExprOp) {
+            super(context, genReference(sourceValues, toExpr), toExpr);
             this.sourceValues = ImmutableList.copyOf(sourceValues);
             this.mergeExprOp = mergeExprOp;
         }
 
+        private static Expression genReference(List<ValueDesc> sourceValues, Expression toExpr) {
```

Comment:
add some comment to explain why need generate reference from `toExpr`

Reply:
> add some comment to explain why need generate reference from `toExpr`

had add comment